src/combinatorics/hales_jewett.lean

```diff
@@ -0,0 +1,299 @@
+/-
+Copyright (c) 2021 David Wärn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Wärn
+-/
+import data.fintype.basic
+import algebra.big_operators.basic
+import data.equiv.fin
+
+/-!
+# The Hales-Jewett theorem
+
+We prove the Hales-Jewett theorem and deduce Van der Waerden's theorem as a corollary.
+
+We consistently work with finite types instead of natural numbers. That is, we write `m, n, k` for
```

Comment:
I appreciate the use of `m,n,k` to reflect the usage of these as naturals but I think in this file it might be nice to use mathlib's usual type-name convention of greek letters (and this docstring can be used to explain why, just as you already have)

Reply:
Yes, I guess it is nice to have the names reflect that these are just generic types